apkid/rules/elf/obfuscators.yara

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2017  RedNaga. http://rednaga.io
+ * All rights reserved. Contact: rednaga@protonmail.com
+ *
+ *
+ * This file is part of APKiD
+ *
+ *
+ * Commercial License Usage
+ * ------------------------
+ * Licensees holding valid commercial APKiD licenses may use this file
+ * in accordance with the commercial license agreement provided with the
+ * Software or, alternatively, in accordance with the terms contained in
+ * a written agreement between you and RedNaga.
+ *
+ *
+ * GNU General Public License Usage
+ * --------------------------------
+ * Alternatively, this file may be used under the terms of the GNU General
+ * Public License version 3.0 as published by the Free Software Foundation
+ * and appearing in the file LICENSE.GPL included in the packaging of this
+ * file. Please visit http://www.gnu.org/copyleft/gpl.html and review the
+ * information to ensure the GNU General Public License version 3.0
+ * requirements will be met.
+ *
+ **/
+
+
+rule ollvm_v3_4 : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM version 3.4 svn"
+    info = "https://github.com/obfuscator-llvm/obfuscator/wiki"
+    // "Obfuscator-clang version 3.4 (tags/RELEASE_34/final) (based on LLVM 3.4svn)"
+
+  strings:
+    $clang_version = "Obfuscator-clang version 3.4"
+    $based_on = "(based on LLVM 3.4svn)"
+
+  condition:
+    all of them
+}
+
+rule ollvm_v3_6_1 : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM version 3.6.1"
+    info = "https://github.com/obfuscator-llvm/obfuscator/wiki"
+    // "Obfuscator-LLVM clang version 3.6.1 (tags/RELEASE_361/final) (based on Obfuscator-LLVM 3.6.1)"
+
+  strings:
+    $clang_version = "Obfuscator-LLVM clang version 3.6.1"
+    $based_on = "(based on Obfuscator-LLVM 3.6.1)"
+
+  condition:
+    all of them
+}
+
+
+rule ollvm_v4_0 : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM version 4"
+    info = "https://github.com/obfuscator-llvm/obfuscator/wiki"
+    // "Obfuscator-LLVM clang version 4.0.1  (based on Obfuscator-LLVM 4.0.1)"
+
+  strings:
+    $clang_version = "Obfuscator-LLVM clang version 4"
+    $based_on = "(based on Obfuscator-LLVM"
+
+  condition:
+    all of them
+}
+
+
+rule ollvm : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM"
+    info = "https://github.com/obfuscator-llvm/obfuscator/wiki"
+
+  strings:
+    $ollvm = "Obfuscator-LLVM"
```

Comment:
What about also adding "(based on " ? This rule is a bit generic as-is.

Reply:
I still think that this string will hunt much more. Also, consider to create two version-less rules as described above in a comment.

Reply:
There's always a trade off between detection and false positives. I think we should minimize false positives. In this case, just this one string would probably false positive and it's possible to write it as a regular expression such that it's much less likely to false positive and wouldn't give up much in detection.

Reply:
Added extra-spaces & add second possible version-less match ( $ollvm1-2 )

```
rule ollvm : obfuscator
{
  meta:
    description = "Obfuscator-LLVM version unknown"
    info        = "https://github.com/obfuscator-llvm/obfuscator/wiki"

  strings:
    $ollvm1 = "Obfuscator-LLVM "
    $ollvm2 = "Obfuscator-clang "

  condition:
    ($ollvm1 or $ollvm2) and
    not ollvm_v3_4 and
    not ollvm_v3_6_1 and
    not ollvm_v4_0 and
    not ollvm_v6_0
}

```